metrics-service/src/main/java/org/zowe/apiml/metrics/security/SpringSecurityConfiguration.java

```diff
@@ -95,10 +94,8 @@ protected void configure(HttpSecurity http) throws Exception {
 
             // login endpoint
             .and()
-            .addFilterBefore(
-                loginFilter(authConfigurationProperties.getServiceLoginEndpoint()),
-                UsernamePasswordAuthenticationFilter.class
-            )
+            .addFilterBefore(new ShouldBeAlreadyAuthenticatedFilter(authConfigurationProperties.getServiceLoginEndpoint(), handlerInitializer.getAuthenticationFailureHandler()), UsernamePasswordAuthenticationFilter.class)
```

Comment:
Why to do it in the Metrics service?

Reply:
Because the Login filter is changed to fall through when there are no credentials, so it does not stop the authentication. Then the filter chain behaves differently. I have to put the ShouldBeAlreadyAuthenticatedFilter after login filter so that the filter chain stops processing.